core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -261,6 +271,21 @@ object ZValidation extends LowPriorityValidationImplicits {
 
   final case class Failure[+W, +E](log: Chunk[W], errors: NonEmptyChunk[E]) extends ZValidation[W, E, Nothing] {
     lazy val errorsUnordered: NonEmptyMultiSet[E] = NonEmptyMultiSet.fromIterable(errors.head, errors.tail)
+
+    lazy val message: String =
+      s"errors:\n  ${errors.map(_.toString).mkString("\n  ")}\nlog:\n  ${log.map(_.toString).mkString("\n  ")}"
+
+    def toException: Failure.Exception[W, E] = Failure.Exception(this)
+
+    def toExceptionWithCause(implicit ev: E <:< Throwable): Failure.Exception[W, E] = {
+      val exn = toException
+      errors.tail.foreach(exn.addSuppressed(_))
```

Comment:
I think if the exception already contains all the errors there is no need to add them as suppressed exceptions here.

Reply:
But "suppressed exceptions" are a standard mechanism on a Throwable that all the ecosystem and tools can with with, `errors` is just our custom field that nobody knows about. Same about "cause".

Reply:
We can just override `getMessage` to render all the failures. That way we can be polymorphic regarding the error type.

Reply:
`ZValidationFailureException` is intended to as much regular JVM exception as possible. The goal is as good interoperability with the rest of the JVM ecosystem and tools as possible. This goes beyond just `message` but also includes `cause`, `suppressed`, etc.